bazel/integration/test_runner/runner.ts

```diff
@@ -38,18 +44,20 @@ export class TestRunner {
     private readonly testPackage: string,
     private readonly toolMappings: Record<string, BazelFileInfo>,
     private readonly npmPackageMappings: Record<string, BazelFileInfo>,
-    private readonly commands: [[binary: string, ...args: string[]]],
+    private readonly commands: [[binary: BazelExpandedValue, ...args: BazelExpandedValue[]]],
+    private readonly environment: Record<string, BazelExpandedValue>,
```

Comment:
Should `TestRunner` just take in a `TestConfig` object rather than separating it out like this and managing the list twice?

Reply:
I'd prefer keeping it that way as the test config was just a way of transferring state from Starlark to JS. Surely this could be the interface for the test runner itself, but subjectively the property access is a a little more convenient that way. We can revisit in the future.

(also we only use the ctor once where the parameters are passed in an obvious way, almost as if they were keyed)

Reply:
Okay, thats fine with me, I just wanted to make sure I pointed it out as its is a bit atypical.  But 👍

Reply:
yeah, definitely a very good point. Looks like both approaches have small trade-offs